In [ ]:
nexflix_cerdits = pd.read_csv('/content/drive/MyDrive/aiffel/Dataton/DataTon/Nexflix/Nexflix_credits.csv')
nexflix_title = pd.read_csv('/content/drive/MyDrive/aiffel/Dataton/DataTon/Nexflix/Nexflix_titles.csv')

In [ ]:
disney_cerdits = pd.read_csv('/content/drive/MyDrive/aiffel/Dataton/DataTon/Disney/Disney_credits.csv')
disney_title = pd.read_csv('/content/drive/MyDrive/aiffel/Dataton/DataTon/Disney/Disney_titles.csv')

In [ ]:
amazon_cerdits = pd.read_csv('/content/drive/MyDrive/aiffel/Dataton/DataTon/Amazon/Amazon_credits.csv')
amazon_title = pd.read_csv('/content/drive/MyDrive/aiffel/Dataton/DataTon/Amazon/Amazon_titles.csv')

In [ ]:
paramount_cerdits = pd.read_csv('/content/drive/MyDrive/aiffel/Dataton/DataTon/Paramount/Paramount_credits.csv')
paramount_title = pd.read_csv('/content/drive/MyDrive/aiffel/Dataton/DataTon/Paramount/Paramount_titles.csv')

In [ ]:
nexflix_cerdits['OTT'] = 'Nexflix'
disney_cerdits['OTT'] = 'Disney'
paramount_cerdits['OTT'] = 'Paramount'
amazon_cerdits['OTT'] = 'Amazon'


In [ ]:
nexflix_title['OTT'] = 'Nexflix'
disney_title['OTT'] = 'Disney'
paramount_title['OTT'] = 'Paramount'
amazon_title['OTT'] = 'Amazon'


In [ ]:
print(nexflix_cerdits.shape)
print('-'*50)
print(disney_cerdits.shape)
print('-'*50)
print(paramount_cerdits.shape)
print('-'*50)
print(amazon_cerdits.shape)

(77213, 6)
--------------------------------------------------
(26412, 6)
--------------------------------------------------
(39842, 6)
--------------------------------------------------
(124235, 6)


In [ ]:
ott_credits = pd.concat([nexflix_cerdits,disney_cerdits,paramount_cerdits,amazon_cerdits])
ott_credits


person_id         id             name                character  \
0            3748    tm84618   Robert De Niro            Travis Bickle   
1           14658    tm84618     Jodie Foster            Iris Steensma   
2            7064    tm84618    Albert Brooks                      Tom   
3            3739    tm84618    Harvey Keitel  Matthew 'Sport' Higgins   
4           48933    tm84618  Cybill Shepherd                    Betsy   
...           ...        ...              ...                      ...   
124230    1938589  tm1054116    Sangam Shukla                   Madhav   
124231    1938565  tm1054116     Vijay Thakur            Sanjay Thakur   
124232     728899  tm1054116    Vanya Wellens                  Budhiya   
124233    1938620  tm1054116     Vishwa Bhanu                    Gissu   
124234    1938620  tm1054116     Vishwa Bhanu                      NaN   

            role      OTT  
0          ACTOR  Nexflix  
1          ACTOR  Nexflix  
2          ACTOR  Nexflix  
3          ACTOR  Nexflix  
4          ACTOR  Nexflix  
...          ...      ...  
124230     ACTOR   Amazon  
124231     ACTOR   Amazon  
124232     ACTOR   Amazon  
124233     ACTOR   Amazon  
124234  DIRECTOR   Amazon  

[267702 rows x 6 columns]

In [ ]:
nexflix_title.rename(columns={'id':'title_id'},inplace = True)
paramount_title.rename(columns={'id':'title_id'},inplace = True)
disney_title.rename(columns={'id':'title_id'},inplace = True)
amazon_title.rename(columns={'id':'title_id'},inplace = True)

In [ ]:
print(nexflix_title.shape)
print('-'*50)
print(disney_title.shape)
print('-'*50)
print(paramount_title.shape)
print('-'*50)
print(amazon_title.shape)

(5806, 16)
--------------------------------------------------
(1535, 16)
--------------------------------------------------
(2825, 16)
--------------------------------------------------
(9871, 16)


In [ ]:
ott_title = pd.concat([nexflix_title,disney_title,paramount_title,amazon_title])
ott_title


title_id                                title   type  \
0      ts300399  Five Came Back: The Reference Films   SHOW   
1       tm84618                          Taxi Driver  MOVIE   
2      tm127384      Monty Python and the Holy Grail  MOVIE   
3       tm70993                        Life of Brian  MOVIE   
4      tm190788                         The Exorcist  MOVIE   
...         ...                                  ...    ...   
9866   tm510327                         Lily Is Here  MOVIE   
9867  tm1079144      Jay Nog: Something from Nothing  MOVIE   
9868   tm847725                              Chasing  MOVIE   
9869  tm1054116                             Baikunth  MOVIE   
9870   ts275838                     Waking Up Eighty   SHOW   

                                            description  release_year  \
0     This collection includes 12 World War II-era p...          1945   
1     A mentally unstable Vietnam War veteran works ...          1976   
2     King Arthur, accompanied by his squire, recrui...          1975   
3     Brian Cohen is an average young Jewish man, bu...          1979   
4     12-year-old Regan MacNeil begins to adapt an e...          1973   
...                                                 ...           ...   
9866  Dallas and heroin have one thing in common: Du...          2021   
9867  Something From Nothing takes you on a stand-up...          2021   
9868  A cop from Chennai sets out to nab a dreaded d...          2021   
9869  This story is about prevalent caste problem, e...          2021   
9870  Kara Stewart, 16, is fed up with just about ev...          2021   

     age_certification  runtime                 genres production_countries  \
0                TV-MA       48      ['documentation']               ['US']   
1                    R      113     ['crime', 'drama']               ['US']   
2                   PG       91  ['comedy', 'fantasy']               ['GB']   
3                    R       94             ['comedy']               ['GB']   
4                    R      133             ['horror']               ['US']   
...                ...      ...                    ...                  ...   
9866               NaN       93              ['drama']               ['US']   
9867               NaN       55             ['comedy']               ['US']   
9868               NaN      116              ['crime']               ['IN']   
9869               NaN       72    ['family', 'drama']                   []   
9870               NaN       10              ['drama']                   []   

      seasons     imdb_id  imdb_score  imdb_votes  tmdb_popularity  \
0         1.0         NaN         NaN         NaN            0.600   
1         NaN   tt0075314         8.3    795222.0           27.612   
2         NaN   tt0071853         8.2    530877.0           18.216   
3         NaN   tt0079470         8.0    392419.0           17.505   
4         NaN   tt0070047         8.1    391942.0           95.337   
...       ...         ...         ...         ...              ...   
9866      NaN   tt7672388         5.3        20.0            1.406   
9867      NaN  tt15041600         NaN         NaN            0.600   
9868      NaN         NaN         NaN         NaN            1.960   
9869      NaN  tt14331982         8.4        49.0            0.645   
9870      1.0  tt13542552         NaN         NaN              NaN   

      tmdb_score      OTT  
0            NaN  Nexflix  
1            8.2  Nexflix  
2            7.8  Nexflix  
3            7.8  Nexflix  
4            7.7  Nexflix  
...          ...      ...  
9866         NaN   Amazon  
9867         NaN   Amazon  
9868         NaN   Amazon  
9869         NaN   Amazon  
9870         NaN   Amazon  

[20037 rows x 16 columns]

In [ ]:
ott_credits.to_csv('/content/drive/MyDrive/aiffel/Dataton/DataTon/Nexflix/ott_credits.csv',index=False)
ott_title.to_csv('/content/drive/MyDrive/aiffel/Dataton/DataTon/Nexflix/ott_title.csv',index=False)

-------

### 1. titles.csv(영상 정보)

- title_ID: 서비스의 타이틀 ID
- title: 제목
- show type: 영상 분류(TV 쇼 or 영화)
- description: 영상의 간략한 설명
- release date: 출시 년도
- age certification: 연령 인증
- runtime: 에피소드(SHOW) 또는 영화의 길이
- genres: 장르 목록
- production countries: 타이틀을 제작한 국가 목록
- Seasons: SHOW인 경우 시즌 수
- IMDB ID: IMDB의 타이틀 ID
- IMDB score: IMDB 점수
- IMDB Votes: IMDB에 대한 투표
- TMDB Popularity: TMDB의 인기도
- TMDB Score: TMDB 점수
- OTT : 기업별 데이터 구분을 위한 구분자

-------
### 2. credits.csv(영상 타이틀별 감독, 배우 등 연출/출연진 정보)

- person ID: 서비스의 개인 ID
- ID: 서비스의 타이틀 ID
- name: 배우 또는 감독 이름
- charecter name : 캐릭터명
- role: ACTOR 또는 DIRECTOR
- OTT : 기업별 데이터 구분을 위한 구분자

-------------
# 전처리 시작

시청등급 / 시즌 / 런타임

In [127]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt 
import seaborn as sns

In [128]:
ott_credits = pd.read_csv('https://raw.githubusercontent.com/saerannn/OTT-Platforms-Analytics/main/src/data/ott_credits.csv?token=GHSAT0AAAAAABTRWD4Y7PRWCR54GDUEK6ZKYUVQGSQ')
ott_credits

person_id         id             name                character  \
0            3748    tm84618   Robert De Niro            Travis Bickle   
1           14658    tm84618     Jodie Foster            Iris Steensma   
2            7064    tm84618    Albert Brooks                      Tom   
3            3739    tm84618    Harvey Keitel  Matthew 'Sport' Higgins   
4           48933    tm84618  Cybill Shepherd                    Betsy   
...           ...        ...              ...                      ...   
267697    1938589  tm1054116    Sangam Shukla                   Madhav   
267698    1938565  tm1054116     Vijay Thakur            Sanjay Thakur   
267699     728899  tm1054116    Vanya Wellens                  Budhiya   
267700    1938620  tm1054116     Vishwa Bhanu                    Gissu   
267701    1938620  tm1054116     Vishwa Bhanu                      NaN   

            role      OTT  
0          ACTOR  Nexflix  
1          ACTOR  Nexflix  
2          ACTOR  Nexflix  
3          ACTOR  Nexflix  
4          ACTOR  Nexflix  
...          ...      ...  
267697     ACTOR   Amazon  
267698     ACTOR   Amazon  
267699     ACTOR   Amazon  
267700     ACTOR   Amazon  
267701  DIRECTOR   Amazon  

[267702 rows x 6 columns]

In [129]:
ott_title = pd.read_csv('https://raw.githubusercontent.com/saerannn/OTT-Platforms-Analytics/main/src/data/ott_title.csv?token=GHSAT0AAAAAABTRWD4ZVTFCF26AMWUSZGHOYUVQHKA')
ott_title

title_id                                title   type  \
0       ts300399  Five Came Back: The Reference Films   SHOW   
1        tm84618                          Taxi Driver  MOVIE   
2       tm127384      Monty Python and the Holy Grail  MOVIE   
3        tm70993                        Life of Brian  MOVIE   
4       tm190788                         The Exorcist  MOVIE   
...          ...                                  ...    ...   
20031   tm510327                         Lily Is Here  MOVIE   
20032  tm1079144      Jay Nog: Something from Nothing  MOVIE   
20033   tm847725                              Chasing  MOVIE   
20034  tm1054116                             Baikunth  MOVIE   
20035   ts275838                     Waking Up Eighty   SHOW   

                                             description  release_year  \
0      This collection includes 12 World War II-era p...          1945   
1      A mentally unstable Vietnam War veteran works ...          1976   
2      King Arthur, accompanied by his squire, recrui...          1975   
3      Brian Cohen is an average young Jewish man, bu...          1979   
4      12-year-old Regan MacNeil begins to adapt an e...          1973   
...                                                  ...           ...   
20031  Dallas and heroin have one thing in common: Du...          2021   
20032  Something From Nothing takes you on a stand-up...          2021   
20033  A cop from Chennai sets out to nab a dreaded d...          2021   
20034  This story is about prevalent caste problem, e...          2021   
20035  Kara Stewart, 16, is fed up with just about ev...          2021   

      age_certification  runtime                 genres production_countries  \
0                 TV-MA       48      ['documentation']               ['US']   
1                     R      113     ['crime', 'drama']               ['US']   
2                    PG       91  ['comedy', 'fantasy']               ['GB']   
3                     R       94             ['comedy']               ['GB']   
4                     R      133             ['horror']               ['US']   
...                 ...      ...                    ...                  ...   
20031            uknown       93              ['drama']               ['US']   
20032            uknown       55             ['comedy']               ['US']   
20033            uknown      116              ['crime']               ['IN']   
20034            uknown       72    ['family', 'drama']                   []   
20035            uknown       10              ['drama']                   []   

       seasons     imdb_id  imdb_score  imdb_votes  tmdb_popularity  \
0          1.0         NaN         NaN         NaN            0.600   
1          1.0   tt0075314         8.3    795222.0           27.612   
2          1.0   tt0071853         8.2    530877.0           18.216   
3          1.0   tt0079470         8.0    392419.0           17.505   
4          1.0   tt0070047         8.1    391942.0           95.337   
...        ...         ...         ...         ...              ...   
20031      1.0   tt7672388         5.3        20.0            1.406   
20032      1.0  tt15041600         NaN         NaN            0.600   
20033      1.0         NaN         NaN         NaN            1.960   
20034      1.0  tt14331982         8.4        49.0            0.645   
20035      1.0  tt13542552         NaN         NaN              NaN   

       tmdb_score      OTT  
0             NaN  Nexflix  
1             8.2  Nexflix  
2             7.8  Nexflix  
3             7.8  Nexflix  
4             7.7  Nexflix  
...           ...      ...  
20031         NaN   Amazon  
20032         NaN   Amazon  
20033         NaN   Amazon  
20034         NaN   Amazon  
20035         NaN   Amazon  

[20036 rows x 16 columns]

'TV-PG': 어린이가 시청하려면 보호자 지도가 권장되는 프로그램

'TV-MA': 17세 미만의 어린이 혹은 청소년한테 부적절한 프로그램'

'TV-Y7-FV': 7세 이상의 어린이를 위한 프로그램으로 매우 경미한 가상의 폭력 묘사가 있음

'TV-Y7': 7세 이상의 어린이를 위한 프로그램

'TV-14': 14세 미만의 어린이 혹은 청소년이 시청하려면 보호자 지도가 권장되는 프로그램

'R': 15세 관람가

'TV-Y': 영유아를 위한 프로그램

'NR': 청소년 관람불가

'PG-13': 12세 관람가에서 15세 관람가

'TV-G': 모든 연령이 시청할 수 있는 프로그램. 다만 어린이를 대상으로 하지는 않았다

'PG': 12세 관람가

'G': 전체관람가

'UR': 청소년 관람불가 등급

'NC-17': 청소년 관람불가 등급

In [130]:
ott_title['age_certification'].values

array(['TV-MA', 'R', 'PG', ..., 'uknown', 'uknown', 'uknown'],
      dtype=object)

In [131]:
df = ott_title.copy()

In [132]:
a = df.groupby('OTT')['age_certification'].value_counts().reset_index(name='count')
a = a.replace('uknown',np.NaN)
a = a.dropna()
a

OTT age_certification  count
1      Amazon                 R   1249
2      Amazon             PG-13    588
3      Amazon                PG    582
4      Amazon                 G    269
5      Amazon             TV-MA    217
6      Amazon             TV-14    188
7      Amazon             TV-PG     91
8      Amazon              TV-Y     78
9      Amazon              TV-G     57
10     Amazon             TV-Y7     52
11     Amazon             NC-17     13
12     Disney                 G    409
13     Disney                PG    354
15     Disney             PG-13     99
16     Disney             TV-PG     94
17     Disney              TV-G     92
18     Disney             TV-Y7     63
19     Disney             TV-14     49
20     Disney              TV-Y     44
21     Disney             TV-MA      6
23    Nexflix             TV-MA    841
24    Nexflix                 R    575
25    Nexflix             TV-14    470
26    Nexflix             PG-13    440
27    Nexflix                PG    246
28    Nexflix             TV-PG    186
29    Nexflix                 G    131
30    Nexflix             TV-Y7    112
31    Nexflix              TV-Y    105
32    Nexflix              TV-G     76
33    Nexflix             NC-17     14
35  Paramount                 R    321
36  Paramount             PG-13    180
37  Paramount             TV-14    165
38  Paramount                PG    147
39  Paramount             TV-PG    143
40  Paramount              TV-G     95
41  Paramount                 G     90
42  Paramount             TV-MA     56
43  Paramount             TV-Y7     52
44  Paramount              TV-Y     50
45  Paramount             NC-17      3

In [133]:
import plotly.express as px 
fig = px.bar(a, x='OTT', y='count', color='age_certification', text = 'count')
fig.show()

해당 값들은 각 국가별로 연령별 지정 값이 다르기 때문에 
ott사별 장르분포가 다양하게 나눠져있다.

장르별 기준치를 전체를 통일할 필요성이 있어보임

-------
- 0 ~ 7세 수준을 kids
- 7 ~ 14세 수준을 children
- 14 ~ 17세 수준을 teenager
- 그 이상을 adult로 치환

In [134]:
ratings_ages = {
    'TV-PG': 'chlidren',
    'TV-MA': 'Adults',
    'TV-Y7-FV': 'chlidren',
    'TV-Y7': 'chlidren',
    'TV-14': 'Teenager',
    'R': 'Teenager',
    'TV-Y': 'Kids',
    'NR': 'Adults',
    'PG-13': 'chlidren',
    'TV-G': 'chlidren',
    'PG': 'chlidren',
    'G': 'Kids',
    'UR': 'Adults',
    'NC-17': 'Adults'
}

ott_title['target_ages'] = ott_title['age_certification'].replace(ratings_ages)
ott_title['target_ages'].unique()

array(['Adults', 'Teenager', 'chlidren', 'Kids', 'uknown'], dtype=object)

In [135]:
df = ott_title.copy()

In [136]:
a = df.groupby('OTT')['target_ages'].value_counts().reset_index(name='count')
a = a.replace('uknown',np.NaN)
a = a.dropna()
a

OTT target_ages  count
1      Amazon    Teenager   1437
2      Amazon    chlidren   1370
3      Amazon        Kids    347
4      Amazon      Adults    230
5      Disney    chlidren    702
6      Disney        Kids    453
8      Disney    Teenager     49
9      Disney      Adults      6
11    Nexflix    chlidren   1060
12    Nexflix    Teenager   1045
13    Nexflix      Adults    855
14    Nexflix        Kids    236
16  Paramount    chlidren    617
17  Paramount    Teenager    486
18  Paramount        Kids    140
19  Paramount      Adults     59

In [137]:
import plotly.express as px 
import plotly.graph_objects as go
fig = px.bar(a, x='OTT', y='count', color='target_ages', text = 'count')

fig.show()

- 4가지 기업 모두 children(7~14) 수준의 장르가 많이 분포 되어있다.

- amazon의 경우 의외로 adult(성인)을 위한 장르가 상대적으로 적으며, 성인 이하를 위한 콘텐츠 들이 다수 분포하는 것을 알 수 있음

- Disney의 경우 데이터셋의 수가 적고, 의외로 teenager(14~17) 수준의 장르의 분포가 적음을 알 수 있다. 또한 adults를 위한 장르가 시각적으로 보이지 않을 만큼 적은 수를 가지고 있다. (실제 수치는 6개)

- Nexflix는 전체적으로 고르게 분포가 되어 있으며, 4가지 기업 중 adults를 위한 장르가 가장 많다. 

- paramount는 17세 이하 즉, 미성년자를 위한 장르가 많으며 adults를 위한 장르는 다소 작다.

-----
# 연령별 장르(rating_age)에 따른 ott사별 점수 평가 비교(imdb기준)

In [138]:
df = ott_title.copy()

In [139]:
a2 = df.groupby(['OTT','target_ages'])['imdb_score'].mean().reset_index(name='mean')
a2 = a2.replace('uknown',np.NaN)
a2 = a2.dropna()
a2

OTT target_ages      mean
0      Amazon      Adults  7.260090
1      Amazon        Kids  6.077287
2      Amazon    Teenager  5.846402
3      Amazon    chlidren  6.126615
5      Disney      Adults  7.683333
6      Disney        Kids  6.401303
7      Disney    Teenager  7.214894
8      Disney    chlidren  6.654112
10    Nexflix      Adults  7.055155
11    Nexflix        Kids  6.466332
12    Nexflix    Teenager  6.738313
13    Nexflix    chlidren  6.510792
15  Paramount      Adults  6.753571
16  Paramount        Kids  5.942400
17  Paramount    Teenager  5.977637
18  Paramount    chlidren  6.470320

In [140]:
a3 = df.groupby(['OTT','target_ages'])['tmdb_score'].mean().reset_index(name='mean')
a3 = a3.replace('uknown',np.NaN)
a3 = a3.dropna()
a3

OTT target_ages      mean
0      Amazon      Adults  7.218468
1      Amazon        Kids  6.220539
2      Amazon    Teenager  5.924073
3      Amazon    chlidren  6.190218
5      Disney      Adults  7.450000
6      Disney        Kids  6.698214
7      Disney    Teenager  8.200000
8      Disney    chlidren  7.004401
10    Nexflix      Adults  7.360749
11    Nexflix        Kids  7.175362
12    Nexflix    Teenager  6.955015
13    Nexflix    chlidren  6.901662
15  Paramount      Adults  7.039286
16  Paramount        Kids  6.457258
17  Paramount    Teenager  6.214004
18  Paramount    chlidren  6.539122

In [141]:
fig = px.bar(a2, x='OTT', y='mean', color = 'target_ages', title = 'mean_IMDB_score by target_ages',text = 'mean')
fig.show()

- amazon = adult(7.2) > chlidren(6.1) > kids(6.0) > teenager(5.8)

- disney = adult(7.68) > teenager(7.2) > chlidren(6.6) > kids(6.4)

- nexflix = adult(7.0) > teenager(6.7) > chlidren(6.5) > kids(6.4)

- paramount = adult(6.7) > chlidren(6.4) > teenager(5.97) > kids(5.94)

In [142]:
fig = px.bar(a3, x='OTT', y='mean', color = 'target_ages',title = 'mean_TMDB_score by target_ages', text = 'mean')
fig.show()

amazon = adult(7.2) > kids(6.2) > children(6.1) > teenager(5.9)

disney = teenager(8.2) > adult(7.45)>  chlidren(7.0) > kids(6.7)

nexflix = adult(7.3) > kids(7.1) > teenager(6.95) > chlidren(6.9)

paramount = adult(7.0) > chlidren(6.5) > kids(6.4) > teenager(6.2)

--------
# 장르별 점

In [143]:
ott_title['genres'].value_counts()

['comedy']                                              1473
['drama']                                               1412
['documentation']                                       1324
[]                                                       358
['drama', 'romance']                                     340
                                                        ... 
['drama', 'romance', 'thriller', 'fantasy']                1
['comedy', 'crime', 'romance', 'thriller', 'action']       1
['scifi', 'war', 'drama']                                  1
['drama', 'fantasy', 'thriller', 'crime']                  1
['thriller', 'western', 'comedy']                          1
Name: genres, Length: 3531, dtype: int64

In [159]:
table = str.maketrans("[]'", '   ')
genres_list = "".join(np.array(ott_title['genres'].str.translate(table)).flatten().tolist())

genres_list = genres_list.replace(',',' ')

import re
genres_list = re.sub(r'[" "]+', " ", genres_list)
genres_list = genres_list.split()
genres_list = set(genres_list)
genres_list = list(genres_list)
genres_list

['horror',
 'drama',
 'action',
 'european',
 'animation',
 'scifi',
 'sport',
 'comedy',
 'war',
 'reality',
 'romance',
 'documentation',
 'crime',
 'family',
 'fantasy',
 'western',
 'history',
 'music',
 'thriller']

In [147]:
for genres in a:
  ott_title[f'{genres}'] = ott_title['genres'].str.contains(genres)

In [158]:
ott_title = ott_title.replace(True,int(1))
ott_title = ott_title.replace(False, int(0))
ott_title

title_id                                title   type  \
0       ts300399  Five Came Back: The Reference Films   SHOW   
1        tm84618                          Taxi Driver  MOVIE   
2       tm127384      Monty Python and the Holy Grail  MOVIE   
3        tm70993                        Life of Brian  MOVIE   
4       tm190788                         The Exorcist  MOVIE   
...          ...                                  ...    ...   
20031   tm510327                         Lily Is Here  MOVIE   
20032  tm1079144      Jay Nog: Something from Nothing  MOVIE   
20033   tm847725                              Chasing  MOVIE   
20034  tm1054116                             Baikunth  MOVIE   
20035   ts275838                     Waking Up Eighty   SHOW   

                                             description  release_year  \
0      This collection includes 12 World War II-era p...          1945   
1      A mentally unstable Vietnam War veteran works ...          1976   
2      King Arthur, accompanied by his squire, recrui...          1975   
3      Brian Cohen is an average young Jewish man, bu...          1979   
4      12-year-old Regan MacNeil begins to adapt an e...          1973   
...                                                  ...           ...   
20031  Dallas and heroin have one thing in common: Du...          2021   
20032  Something From Nothing takes you on a stand-up...          2021   
20033  A cop from Chennai sets out to nab a dreaded d...          2021   
20034  This story is about prevalent caste problem, e...          2021   
20035  Kara Stewart, 16, is fed up with just about ev...          2021   

      age_certification  runtime                 genres production_countries  \
0                 TV-MA       48      ['documentation']               ['US']   
1                     R      113     ['crime', 'drama']               ['US']   
2                    PG       91  ['comedy', 'fantasy']               ['GB']   
3                     R       94             ['comedy']               ['GB']   
4                     R      133             ['horror']               ['US']   
...                 ...      ...                    ...                  ...   
20031            uknown       93              ['drama']               ['US']   
20032            uknown       55             ['comedy']               ['US']   
20033            uknown      116              ['crime']               ['IN']   
20034            uknown       72    ['family', 'drama']                   []   
20035            uknown       10              ['drama']                   []   

       seasons  ... reality  romance  documentation  crime  family fantasy  \
0          1.0  ...       0        0              1      0       0       0   
1          1.0  ...       0        0              0      1       0       0   
2          1.0  ...       0        0              0      0       0       1   
3          1.0  ...       0        0              0      0       0       0   
4          1.0  ...       0        0              0      0       0       0   
...        ...  ...     ...      ...            ...    ...     ...     ...   
20031      1.0  ...       0        0              0      0       0       0   
20032      1.0  ...       0        0              0      0       0       0   
20033      1.0  ...       0        0              0      1       0       0   
20034      1.0  ...       0        0              0      0       1       0   
20035      1.0  ...       0        0              0      0       0       0   

      western  history  music  thriller  
0           0        0      0         0  
1           0        0      0         0  
2           0        0      0         0  
3           0        0      0         0  
4           0        0      0         0  
...       ...      ...    ...       ...  
20031       0        0      0         0  
20032       0        0      0         0  
20033       0        0      0         0  
20034       0        0      0   

In [240]:
df = ott_title.groupby('OTT').sum()
df

release_year  runtime  seasons  imdb_score   imdb_votes  \
OTT                                                                  
Amazon         19755101   848640  12302.0     52891.1   75437150.0   
Disney          3075492    92351   2192.0      7318.5   84366796.0   
Nexflix        11702959   450790   8191.0     34516.2  123285696.0   
Paramount       5623728   204120   4691.0     15790.0   58109868.0   

           tmdb_popularity  tmdb_score  horror  drama  action  ...  reality  \
OTT                                                            ...            
Amazon        64430.745997     46611.3    1065   4764    1820  ...      138   
Disney        61323.453000      9862.7      29    347     304  ...       62   
Nexflix      128666.568354     37417.4     380   2901    1053  ...      223   
Paramount     36105.344442     14128.4     259   1072     533  ...      109   

           romance  documentation  crime  family  fantasy  western  history  \
OTT                                                                           
Amazon        1752           1096   1251     751      554      490      396   
Disney         138            351     47     754      392       23       41   
Nexflix        958            910    891     622      631       44      233   
Paramount      490            495    359     286      181      231      126   

           music  thriller  
OTT                         
Amazon       438      2119  
Disney        98        57  
Nexflix      238      1178  
Paramount    146       511  

[4 rows x 26 columns]

In [241]:
df = df.iloc[:,7:]
df['total'] = df.sum(axis=1)
df = df.reset_index()
df

OTT  horror  drama  action  european  animation  scifi  sport  \
0     Amazon    1065   4764    1820       712        475    705    228   
1     Disney      29    347     304        35        486    265     63   
2    Nexflix     380   2901    1053       460        665    587    166   
3  Paramount     259   1072     533       101        176    244     51   

   comedy  war  ...  romance  documentation  crime  family  fantasy  western  \
0    2987  324  ...     1752           1096   1251     751      554      490   
1     649    6  ...      138            351     47     754      392       23   
2    2269  149  ...      958            910    891     622      631       44   
3    1007  113  ...      490            495    359     286      181      231   

   history  music  thriller  total  
0      396    438      2119  22065  
1       41     98        57   4147  
2      233    238      1178  14558  
3      126    146       511   6490  

[4 rows x 21 columns]

In [245]:
df.columns

Index(['OTT', 'horror', 'drama', 'action', 'european', 'animation', 'scifi',
       'sport', 'comedy', 'war', 'reality', 'romance', 'documentation',
       'crime', 'family', 'fantasy', 'western', 'history', 'music', 'thriller',
       'total'],
      dtype='object')

In [249]:
# fig = px.bar(df, x='OTT', y=df.sum(axis=1), color = )
# fig.show()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning:

Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.

